src/groupeddataframe/splitapplycombine.jl

```diff
@@ -414,10 +416,45 @@ julia> combine(df -> sum(df.c), gd) # Slower variant
 
 See [`by`](@ref) for more examples.
 """
-function combine(f::Any, gd::GroupedDataFrame)
+combine(f::Function, gd::GroupedDataFrame; keepkeys::Bool=true) =
+    combine_helper(f, gd, keepkeys=keepkeys)
+combine(gd::GroupedDataFrame, f::Function; keepkeys::Bool=true) =
+    combine_helper(f, gd, keepkeys=keepkeys)
+
+function DataFrames.combine(gd::GroupedDataFrame, cs::Pair...; keepkeys::Bool=true)
+    if any(x -> first(x) isa Tuple, cs)
+        Base.depwarn("passing a Tuple as column selector is deprecated, use " *
+                     "an AbstractVector instead", :combine)
+        cs = collect(cs)
+        for (i, v) in enumerate(cs)
+            if first(v) isa Tuple
+                cs[i] = collect(first(v)) => last(v)
+            end
+        end
+    end
+    cs_norm = [normalize_selection(index(parent(gd)), c) for c in cs]
+    inverse_cs_norm = [to => (from =>fun) for (from, (fun, to)) in cs_norm]
+    @assert !isempty(inverse_cs_norm)
+    nt_cs = (; inverse_cs_norm...)
```

Comment:
Why build a named tuple here? AFAICT `combine_helper` will pass it to `_combine`, which will collect it to a vector before calling another `_combine` method. Better get rid of named tuples completely.

I also wonder whether `combine_helper` is really needed: couldn't it just be the main `combine` method, with others calling it? All this zoo of "combine" variants is already hard to follow... :-D

Reply:
`combine_helper` is needed for a common code that is reused on `combine` level (like - we do not call `_combine` if `gd` has 0 elements).

I have removed the named tuples in the code (we have to pass around column names though).